status/raw/status-app-protocols.md

```diff
@@ -0,0 +1,372 @@
+
+## Waku transport layer
+
+The Waku transport layer contains the functions allowing Status protocols to use [10/WAKU2](../../waku/standards/core/10/waku2.md) infrastructure as transport.
+
+### Waku messages
+
+Each Status application message MUST be transformed to a [14/WAKU2-MESSAGE](../../waku/standards/core/14/message.md) with the following structure:
+
+```protobuf
+syntax = "proto3";
+
+message WakuMessage {
+  bytes payload = 1;
+  string content_topic = 2;
+  optional uint32 version = 3;
+  optional sint64 timestamp = 10;
+  optional bytes meta = 11;
+  optional bool ephemeral = 31;
+}
+```
+
+- `payload` MUST be set to the full encrypted payload received from the higher layers
+- `version` MUST be set to `1`
+- `ephemeral` MUST be set to `true` if the app-level message is ephemeral
+- `content_topic` MUST be set to the app-level content topic
+- `timestamp` MUST be set to the current Unix epoch timestamp (in nanosecond precision)
+
+### Pubsub topics and sharding
+
+All Waku messages are published to pubsub topics as defined in [23/WAKU2-TOPICS](../../waku/informational/23/topics.md).
+Since pubsub topics define a routing layer for messages,
+they can be used to shard traffic.
+The pubsub topic used for publishing a message depends on the app-level [functional scope](#functional-scope).
+
+**Self-addressed messages**
+
+The application MUST define at least one distinct pubsub topic for self-addressed messages.
+The application MAY defined a set of more than one pubsub topic for self-addressed messages to allow traffic sharding for scalability.
+
+**Global messages**
+
+The application MUST define at least one distinct pubsub topic for global control messages and global content messages.
+The application MAY defined a set of more than one pubsub topic for global messages to allow traffic sharding for scalability.
+It is RECOMMENDED that separate pubsub topics be used for global control messages and global content messages.
+
+**Community messages**
+
+The application SHOULD define at least one separate pubsub topic for each separate community's community control and community content messages.
+The application MAY define a set of more than one pubsub topic per community to allow traffic sharding for scalability.
+It is RECOMMENDED that separate pubsub topics be used for global control messages and global content messages.
+
+**Large messages**
+
+The application MAY define separate pubsub topics for large messages.
+These pubsub topics for large messages MAY be distinct for each functional scope.
+
+### Resource usage
+
+The application can use a range of Waku protocols to interact with the Waku transport layer.
+The specific set of Waku protocols used depend on desired functionality and resource usage profile for the specific client.
+Resources can be restricted in terms of bandwidth and computing resources.
+
+Waku protocols that are more appropriate for resource-restricted environments are often termed "light protocols".
```

Comment:
I have started to write an explanation article about this. Will share draft when ready and we can start to move away from non-relay/light protocol terminology.

Reply:
Yes. I'm happy to also adapt to whatever terminology we deem best. In case it helps, reasoning behind my sticking with "light" and "full":
- people seemed to always revert to this, even while we pushed for "resource-restricted" and "adaptive"
- it fits current Status client terminology
- relay vs non-relay wouldn't be accurate as each client will (or should) use a combination of filter/lightpush/relay etc. depending on resources.